src/main/java/org/broadinstitute/hellbender/engine/dataflow/ReadsSource.java

```diff
@@ -37,19 +35,16 @@
 
     /**
      * @param bam A file path or a google bucket identifier to a bam file to read
-     * @param clientSecret a path to a local client secret file to use to authenticate with gcloud, ignored if bam is a
-     *                     a local file
+     * @param p the pipeline object for the job. This is needed to read a bam from a bucket.
+     *          The options inside of the pipeline MUST BE GCSOptions (to get the secret file).
      */
-    public ReadsSource(String bam, File clientSecret){
+    public ReadsSource(String bam, Pipeline p){
         this.bam = Utils.nonNull(bam);
 
         cloudStorageUrl = BucketUtils.isCloudStorageUrl(bam);
         if(cloudStorageUrl) {
-            if (clientSecret == null || !clientSecret.exists()) {
-                throw new UserException("You must specify a valid client secret file if using bams from a google bucket");
-            }
-            //HACK this is gross but it seemed like the easiest way to deal with the auth stuff
-            options = PipelineOptionsFactory.fromArgs(new String[]{"--genomicsSecretsFile=" + clientSecret.getAbsolutePath()}).as(GCSOptions.class);
+            // The options used to create the pipeline must be GCSOptions to get the secret file.
+            options = p.getOptions().as(GCSOptions.class);
```

Comment:
How about a try/catch for the potential ClassCastException? If you get one, throw a `GATKException`